QGA-20 check permission for quiz and user patch and delete routes

nondante opened the pull request:
Permission check for quiz and user patch and delete routes created.

Commit:
REFFERENCE: QGA-20
DESCRIPTION: Permission check for updateQuiz created

Commit:
REFFERENCE: QGA-20
DESCRIPTION: permission for quiz and user patch and delete routes created

Gruseckis commented on models/UserModel.js:
```diff
@@ -29,6 +29,7 @@ const UserModel = mongoose.model('User',userSchema);
 
 const save = async model => new UserModel(model).save();
 const getUserByUsername = async username => UserModel.findOne({username});
+const getUserById = async id => UserModel.findOne({_id: id});
```
If you are using ID you should probably use findById

Gruseckis commented on controllers/userController.js:
```diff
@@ -46,19 +49,33 @@ const updateOneUser = async (req, res, next) => {
-      });
-    } else {
-      throw new AppError('user not found');
+    if(req.user.level == "admin"){
```
Use triple equal instead of two

Gruseckis commented on controllers/quizController.js:
```diff
@@ -31,30 +31,51 @@ const addQuiz = async (req, res, next) => {
   }
 }
 
+const isOwner = async(quizId, userId) => {
```
Please keep your code DRY

Gruseckis commented on controllers/quizController.js:
```diff
@@ -31,30 +31,51 @@ const addQuiz = async (req, res, next) => {
-    res.status(200).send({
-      payload: updatedQuiz
-    });
+    if(await isOwner(req.params.quizId, req.user._id) || req.user.level=="moderator" || req.user.level=="admin"){
```
Need to refactor this. Also use tripple. There should not be an await in If statement

Gruseckis commented on controllers/quizController.js:
```diff
@@ -31,30 +31,51 @@ const addQuiz = async (req, res, next) => {
-    res.status(200).send({
-      payload: {
-        message: 'Successfully deleted quiz'
+    if(await isOwner(req.params.quizId, req.user._id) || req.user.level=="moderator" || req.user.level=="admin"){
```
Same as the above comment.

Commit:
REFFERENCE: QGA-20
DESCRIPTION: permission for quiz and user patch and delete routes created

Commit:
REFFERENCE: QGA-20
DESCRIPTION: permission for quiz and user patch and delete routes created

Gruseckis commented on controllers/quizController.js:
```diff
@@ -31,30 +32,59 @@ const addQuiz = async (req, res, next) => {
-    res.status(200).send({
-      payload: updatedQuiz
-    });
+    if(await isOwner(req.params.quizId, req.user._id) || accessLevelCheck(req.user.level,'moderator')){
```
remove await from if statement

Gruseckis commented on controllers/quizController.js:
```diff
@@ -31,30 +32,59 @@ const addQuiz = async (req, res, next) => {
-    res.status(200).send({
-      payload: {
-        message: 'Successfully deleted quiz'
+    if(await isOwner(req.params.quizId, req.user._id) || accessLevelCheck(req.user.level,'moderator')){
```
remove await from if

Gruseckis commented on helpers/accessLevelCheck.js:
```diff
@@ -0,0 +1,12 @@
+
+
```
Remove extra white lines.

Commit:
REFFERENCE: QGA-20
DESCRIPTION: Permission check for quiz and user patch and delete routes created.

Commit:
REFFERENCE: QGA-20
DESCRIPTION: findById added to UserModel

Commit:
Fixed getUserById

Gruseckis committed:
Merge branch 'development' into QGA-20-Check-permission-for-quiz-and-user-patch-and-delete-routes
